tests/MongoDB.Driver.Tests/Specifications/transactions/tests/pin-mongos.json

```diff
@@ -875,7 +875,7 @@
                 "failCommands": [
                   "commitTransaction"
                 ],
-                "errorCode": 50
+                "errorCode": 51
```

Comment:
This is the only real change in the scope of this ticket

Reply:
This test didn't fail in the previous implementation because the thrown exception contains `UnknownTransactionCommitResult` label, but according to json file, we're asserting on the exception doesn't contain `TransientTransactionError` label. That's why the test hasn't failed before.

Reply:
If in the previous test we were throwing an `UnknownTransactionCommitResult`, shouldn't the test have failed because an `UnknownTransactionCommitResult` should result in unpinning, according to the description in https://jira.mongodb.org/browse/DRIVERS-764?

Reply:
it didn't fail because we asserted only this label: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/89/files/fbcf63ce7b99df380b432cf5c17c79e6f65ad7ea#diff-78c2fbd4712982e063ddbc76237a1f61R888
So, it's how the json test was designed

Reply:
Shouldn't we also be asserting on that the session is pinned here? https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/89/files/fbcf63ce7b99df380b432cf5c17c79e6f65ad7ea#diff-78c2fbd4712982e063ddbc76237a1f61R894?
In other words, when we had an error code 50, according to the drivers ticket, we should have been unpinning, but if this test was passing, that means we were not unpinning.
If we were to change the errorCode back to 50, does the test pass  on 4.2/latest? If so, we may need to file a new ticket because that feels like a bug, based off my reading of the drivers ticket: what do you think?

Reply:
Per offline discussion, the fact that this test _should_ have been failing is irrelevant to this ticket. The tests currently pass, and I'll wait to hear back from Oleg/Shane to see if a new spec ticket should be filed with a new test